src/doc/man/cargo-yank.md

```diff
@@ -15,9 +15,26 @@ The yank command removes a previously published crate's version from the
 server's index. This command does not delete any data, and the crate will
 still be available for download via the registry's download link.
 
-Note that existing crates locked to a yanked version will still be able to
-download the yanked version to use it. Cargo will, however, not allow any new
-crates to be locked to any yanked version.
+However, yanking a release will prevent cargo from selecting that version
+when determining the version of a dependency to use. If there are no longer
+any compatible versions that haven't been yanked, cargo will return an error.
+
+The only exception to this is crates locked to a specific version by a lockfile,
+these will still be able to download the yanked version to use it.
```

Comment:
```suggestion
Best practice is to yank a release *only* when it contains a serious flaw, such as 
a security vulnerability, and when there is a newer semver compatible release 
published. You do not need to yank a version to suggest users of your crate upgrade. 

Cargo will not use a yanked version for any new project or checkout without a 
pre existing lockfile, and will generate an error if there are no longer
any compatible versions for your crate.

For example, the `foo` crate published version `0.22.0` and another crate `bar`
declared a dependency on version `foo = 0.22`. Now `foo` releases a new, but 
not semver compatibile, version `0.23.0`, and finds a critical issue with `0.22.0`.
If `0.22.0` is yanked, no new project or checkout without an existing lockfile will be
able to use crate `bar` as it relies on `0.22`

In this case, the maintainers of `foo` should first publish a semver compatible version
such as `0.22.1` prior to yanking `0.22.0` so that `bar` and all projects that depend 
on `bar` will continue to work
```

Reply:
> Cargo will not use a yanked version for any new project

I think, that the concept of "new" requires clarification. When project is considered as "new"?

Reply:
Practically speaking, I think "new" means any rust project that does not already have a `Cargo.lock` file. 

I believe the basic guidance is that "libraries" (e.g. stuff published to crates.io that other applications can depend on) should not have a Cargo.lock: https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
> such as a security vulnerability

I think accidentally breaking semver compatibility might deserve mention there as well it can wreak havoc in downstream projects.

Reply:
Probably that explanation also should appear when you try to click "Yank" button on crates.io or run `cargo yank`

Reply:
Updated with a slightly tweaked version of this, PTAL